src/components/GestureButtons.tsx

```diff
@@ -1,65 +1,108 @@
-/* eslint-disable eslint-comments/no-unlimited-disable */
-/* eslint-disable */
-// @ts-nocheck TODO(TS) provide types
-import PropTypes from 'prop-types';
 import React from 'react';
-import { Animated, Platform, processColor, StyleSheet } from 'react-native';
+import {
+  Animated,
+  Platform,
+  processColor,
+  StyleSheet,
+  StyleProp,
+  ViewStyle,
+} from 'react-native';
 
 import createNativeWrapper from '../handlers/createNativeWrapper';
 import GestureHandlerButton from './GestureHandlerButton';
 import State from '../State';
 
-export const RawButton = createNativeWrapper(GestureHandlerButton, {
-  shouldCancelWhenOutside: false,
-  shouldActivateOnStart: false,
-});
+import {
+  GestureEvent,
+  HandlerStateChangeEvent,
+} from '../handlers/gestureHandlers';
+import {
+  NativeViewGestureHandlerPayload,
+  NativeViewGestureHandlerProperties,
+} from '../handlers/NativeViewGestureHandler';
+
+/* BUTTONS PROPERTIES */
+
+interface RawButtonProperties extends NativeViewGestureHandlerProperties {
+  exclusive?: boolean;
+  testID?: string;
+  accessibilityLabel?: string;
+  // TODO: apply proper value
+  rippleColor?: any; // it was present in BaseButtonProperties before but is used here in code
+}
 
-/* Buttons */
+export interface BaseButtonProperties extends RawButtonProperties {
+  onPress?: (pointerInside: boolean) => void;
+  onActiveStateChange?: (active: boolean) => void;
+  style?: StyleProp<ViewStyle>;
+}
 
-export class BaseButton extends React.Component {
-  static propTypes = {
-    ...RawButton.propTypes,
-    onPress: PropTypes.func,
-    onActiveStateChange: PropTypes.func,
-  };
+export interface RectButtonProperties extends BaseButtonProperties {
+  underlayColor?: string;
+  activeOpacity?: number;
+}
+
+export interface BorderlessButtonProperties extends BaseButtonProperties {
+  borderless?: boolean;
+  activeOpacity?: number;
+}
```

Comment:
`activeOpacity?: number;` is repeated in both interfaces, can we move it to `BaseButtonProperties` or not really?

Reply:
I think not really, based on the code, these are `props` intended for these components only.